src/main/java/com/yahoo/validatar/report/email/TestSuiteModel.java

```diff
@@ -0,0 +1,84 @@
+package com.yahoo.validatar.report.email;
+
+import com.yahoo.validatar.common.Query;
+import com.yahoo.validatar.common.Test;
+import com.yahoo.validatar.common.TestSuite;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * This class holds the necessary information for rendering
+ * the Validatar report using Jtwig.
+ */
+public class TestSuiteModel {
+    /**
+     * Test suite name.
+     */
+    public final String name;
+    /**
+     * Number of passed queries.
+     */
+    public final int queryPassed;
+    /**
+     * Total number of queries.
+     */
+    public final int queryTotal;
+    /**
+     * Number of passed tests.
+     */
+    public final int testPassed;
+    /**
+     * Total number of tests.
+     */
+    public final int testTotal;
+
+    /**
+     * List of failed queries.
+     */
+    public final List<Query> failedQueries;
+    /**
+     * List of failed tests.
+     */
+    public final List<Test> failedTests;
+
+    /**
+     * Create a {@code TestSuiteModel} from a {@code TestSuite}.
+     * The constructor will pull the required information from
+     * the given test suite.
+     *
+     * @param testSuite The test suite object to model.
+     */
+    protected TestSuiteModel(TestSuite testSuite) {
+        failedQueries = new LinkedList<>();
+        failedTests = new LinkedList<>();
+        this.name = testSuite.name;
+        int passCount = 0;
+        for (Query query : testSuite.queries) {
+            if (query.failed()) {
+                failedQueries.add(query);
+            } else {
+                passCount++;
+            }
+        }
+        this.queryPassed = passCount;
+        this.queryTotal = testSuite.queries.size();
+        passCount = 0;
+        for (Test test : testSuite.tests) {
+            if (test.failed()) {
```

Comment:
We don't want to mark a test as failed if it was warnOnly. Switching to storing passedTests and using ```test.passed()``` should fix it. We should still note in the email that warned or skipped tests were present. Do you think adding that as a third category or just marking skipped tests as yellow/orange instead of green is better?

Reply:
The template checks for 'failed' tests that are 'warnOnly' and sticks in a <span style="color:red"><b>SKIPPED</b></span> in orange in the report.

Reply:
But it will still display messages, should skipped tests not display their messages?

Reply:
I saw that but correct me if I am wrong, it counts as a failed test right? Basically, we don't want to alert anyone if warnOnly is true because they have asked to warn only.

Displaying messages is totally fine and needed.

Reply:
so basically if there are warn only tests it should still appear as successful?

Reply:
Yea or you can make a new category entirely for successful, skipped and failed tests. I like the third option personally since it's clear.

Reply:
aight